### DISCUSSION TITLE
Recording wav file to a built-in SD card from INMP441 microphone (ESP32 STEAMakers)

### DISCUSSION THREAD

**Discussion by FGAC1226** [pschatzmann/arduino-audio-tools#548] [Q&A] [answered]
I'm using this board https://shop.innovadidactic.com/es/standard-placas-shields-y-kits/1567-placa-esp32-steamakers.html , and I fail to get even the I2S to Serial example to work. I assumed that is because the pins that the INMP441 is supposed to be connected to (SD --> 32, WS --> 15) are unavaliable in this board (because they are dedicated to sd card, I think).
I was wondering if there was a solution to avoid those pins, can I just connect them to whichever? If so, how? 

As for the sd card, I can't find how to use the built-in SD Card as an audio sink.

Any help on those would be appreciated, thanks.

**Comment by pschatzmann** [comment, marked as answer]
I2S on the ESP32 is flexible and you can map the functionality to any other pins. You can do this in the sketch using [I2SConfig](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_i2_s_config.html) or for all sketches you can change the defaults pins in the AudioConfig.h.

For the SD Card the documentation is a little bit confusing: Usually the SPI pins are called
- MISO
- MOSI
- SCLK
- CS or SS

Your documentation however gives
- IO32 D0 – Tarjeta SD (=MISO?)
- IO15 CLK – Tarjeta  (=SCLK)
- IO33 SDCMD – Tarjeta (MOSI?)

I am also confused that the CS is not specified, so you might want to try any unused pin to keep the API happy.
<code>
#define PIN_SD_CARD_CS 5
#define PIN_SD_CARD_MISO 32
#define PIN_SD_CARD_MOSI 33
#define PIN_SD_CARD_CLK  15
// setup SPI for SD card
SPI.begin(PIN_SD_CARD_CLK, PIN_SD_CARD_MISO, PIN_SD_CARD_MOSI, PIN_SD_CARD_CS);
</code>
with this you can Google any SD Arduino tutorial...
